android_kernel_tools.rc

```diff
@@ -673,20 +674,26 @@ function bootzip()
   if [ "${PushToDevice}" = 'true' ]; then
     echo '';
 
-    # Waiting for adb device
+    # Wait for device
     timeout 10 ${adb} 'wait-for-device';
 
-    # Push to common '/sdcard/' path
-    echo "  bootzip: Pushing to device /sdcard/${OutFile}";
-    timeout 10 ${adb} 'wait-for-device';
-    adb push "${TargetFile}" '/sdcard/';
+    # Check if '/sdcard' exists
+    if [ "$(${adb} shell 'test -d /sdcard/ 2> /dev/null; echo $?')" -eq '0' ]; then
```

Comment:
I'd prefer the previous -z "$(adb shell 'ls ${OutFile} 2> /dev/null')" instead of a test and $? output.